dbclient/test/MLFlowClientTest.py

```diff
@@ -0,0 +1,127 @@
+import unittest
+import os
+import sqlite3
+from dbclient import MLFlowClient
+import json
+import concurrent.futures
+from mlflow.entities import Metric, Param, RunTag, RunData, RunInfo, Run
+
+MLFLOW_TEST_FILE = "dbclient/test/mlflow_runs_test.db"
+
+class MLFlowClientTest(unittest.TestCase):
+
+    @classmethod
+    def tearDownClass(cls):
+        if os.path.exists(MLFLOW_TEST_FILE):
+            os.remove(MLFLOW_TEST_FILE)
+
+    def _generate_run(self, i, runs_dict):
+        """
+        Generate a run object and save to runs_dict keyed by run_id.
+        Most of data just depends on i, and some data are hard-coded for simplicityGenerate n number of runs. Most of
+        data just depends on n, and some data are hard-coded for simplicity.
+        """
+        key = f"key{i}"
+        value = f"value{i}"
+        start_time = 123456 * i
+        end_time = start_time + (1000 * i)
+        run_id = f"run_id_{i}"
+
+        metrics = [Metric(key, value, start_time, "stage")]
+        params = [Param(key, value)]
+        tags = [RunTag(key, value)]
+        run_info = RunInfo(run_id, "experiment_id", "user_id", "status", start_time, end_time, "lifecycle_stage")
+        run = Run(
+                    run_info=run_info,
+                    run_data=RunData(
+                    metrics=metrics,
+                    params=params,
+                    tags=tags))
+        runs_dict[run_id] = run
+        return run
+
+    def _insert_run_data(self, run):
+        con = sqlite3.connect(MLFLOW_TEST_FILE, timeout=30)
+        with con:
+            MLFlowClient._save_run_data_to_sql(con, run)
+
+
+    def test_save_run_data_to_sql(self):
```

Comment:
Could we also add a test that proves the idempotency of `MlflowClient._save_run_data_to_sql`, i.e. in case we fail to export an experiment's runs partway and so retry exporting its runs + re-export some runs?

Reply:
It looks like it's already idempotent due to the `REPLACE INTO` syntax, but just thinking a test would be nice to have here

Reply:
good point. added the test!